### ISSUE TITLE
Keep Windows 8.1/Windows Phone 8.1 as a separate target platform from Windows 10?

### ISSUE THREAD

**Issue by brodybits** [cordova/cordova-discuss#34]
Just an idea that would help app developers deal with plugins that are not (yet) working in a real Windows 10 project (such as the sqlite plugin that I maintain).

**Comment by nikhilkh** [comment]
Currently, `windows-target-version` in config.xml determines if you end up building for Windows 8.1 or Windows 10. From our experience in working with core plugins - there were not too many significant differences in being able to run in Windows 8.1/Windows 10. 

I'm curious - What issue are you running into with SQLite plugin?

**Comment by brodybits** [comment]
Most Windows plugins are much easier since they are simply using Javascript proxy code to access built-in Windows features. But SQLite(3) should not be considered built-in so consistently. It is my understanding that someone has to install a nuget package or something to get SQLite(3), and it would probably be a snapshot that is not updated so so often.

I was able to get the SQLite plugin working for Windows 8.1 and Windows Phone 8.1 by using https://github.com/doo/SQLite3-WinRT but with my own custom project files. I used https://github.com/AllJoyn-Cordova/cordova-plugin-alljoyn as an example to embed a C++ library that can be built for both Windows 8.1 and Windows Phone 8.1. (Shout out of thanks to @vjrantal on the @Microsoft team.)

Come Windows 10, I had to revisit the changes and was able to get another custom project file to build for Windows 10. (Fortunately I was able to get the Windows 8.1, Windows Phone 8.1, and Windows 10 project files to use a common project file that defines which C and C++ files to include in the build.)

Now I have some troubles supporting both Windows 8.1/Windows Phone 8.1 and Windows 10 at the same time. This is why it would be easier if we could treat Windows 8.1/Windows Phone 8.1 and Windows 10 as separate platforms.

**Comment by nikhilkh** [comment]
Good to hear that you found a way to share large chuck of the project system for Windows 8.1 & 10. What trouble are you having in supporting both Windows 8.1/Windows Phone 8.1 & Windows 10? Have you reviewed the `versions` attribute that we added to the framework element: https://cordova.apache.org/docs/en/latest/plugin_ref/spec.html#framework
We designed it for plugins to account for such differences using that attribute.
